app/com/yetu/oauth2provider/oauth2/handlers/AuthorizationHandler.scala

```diff
@@ -37,8 +37,8 @@ class AuthorizationHandler(authAccessService: IAuthCodeAccessTokenService,
         logger.debug(s"id = ${oauthClient.clientId} , secret = ${oauthClient.clientSecret}, grantType = ${oauthClient.grantTypes}")
 
         val validClientId = oauthClient.clientId == clientCredential.clientId
-        val validGrantType = oauthClient.grantTypes.map(grantList => grantList.contains(grantType)).getOrElse(false)
-        val validSecret = clientCredential.clientSecret.map(_ == oauthClient.clientSecret).getOrElse(false)
+        val validGrantType = oauthClient.grantTypes.exists(grantList => grantList.contains(grantType))
+        val validSecret = clientCredential.clientSecret.contains(oauthClient.clientSecret)
 
```

Comment:
comparing for equality and comparing whether one is a substring of the other is not quite the same thing. Why did you change this code?